### PR TITLE
[Mantis 0022554] Fix wrong absolute paths (windows)

### PR DESCRIPTION
**Pull request by d3r1w**
This PR provides a possible solution for the issue described in the mantis ticket [0022554](https://www.ilias.de/mantis/view.php?id=22554)

There were two issues causing this problem:
* Wrong path separator between the base path of the filesystem abstraction and the supplied relativ path.
* Different fopen behaviour under windows let to an access denied error while accessing a directory.

Solution:
* Fallback to ILIAS defined constants to build the absolute path in the old filesystem abstraction.
* Replacement of the wrong path separator in the local filesystem abstraction.

### CODE REVIEW

File: src/Filesystem/Provider/FlySystem/FlySystemLocalFilesystemFactory.php [hunk 1]
```diff
@@ -50,6 +53,18 @@ public function getInstance(LocalConfig $config) {
 			]
 			);
 
+		//switch the path separator to a forward slash
+		$reflaction = new \ReflectionObject($adapter);
```
Comment (chfsx) [inline src/Filesystem/Provider/FlySystem/FlySystemLocalFilesystemFactory.php]: Typo in `$reflaction`--> `$reflection`

File: src/Filesystem/Provider/FlySystem/FlySystemLocalFilesystemFactory.php [hunk 2]
```diff
@@ -6,6 +6,9 @@
 use ILIAS\Filesystem\Filesystem;
 use ILIAS\Filesystem\Provider\Configuration\LocalConfig;
 use League\Flysystem\Adapter\Local;
+use ReflectionClass;
```
Comment (chfsx) [inline src/Filesystem/Provider/FlySystem/FlySystemLocalFilesystemFactory.php]: Is `ReflectionClass` needed?

**Review by chfsx — CHANGES_REQUESTED** [review]

**Comment by d3r1w** [comment]
Thx for the quick review @chfsx

**Review by chfsx — APPROVED** [review]

**Comment by chfsx** [comment]
Thank you very much @d3r1w !

**Pull request merged**